SC_redesign/Crop_and_Soil/crop/transpiration.py

```diff
@@ -0,0 +1,304 @@
+            the adjusted potential water to be taken up from each layer (mm)
+
+        Raises
+        ------
+        Exception
+            If the lengths of `potential_uptakes` and `unmet_demands` are not equal.
+        """
+        if not len(potential_uptakes) == len(unmet_demands):
+            raise Exception("potential_uptakes and unmet_demands must be the same length.")
+
+        adjusted = [uptake + (demand * uptake_compensation) for uptake, demand in zip(potential_uptakes, unmet_demands)]
+
+        return adjusted
+
+    @staticmethod
+    def _find_stratified_max_water_uptakes(root_depth: float, max_transpiration: float, water_distro_parameter: float,
+                                           upper_depths: List[float], lower_depths: List[float]) -> List[float]:
+        """calculates the crop's maximum water uptake from each soil layer during the current day.
+
+        SWAT Equation: 5:2.2.2
+
+        Parameters
+        ----------
+        root_depth : float
+            the current depth of the crop root development (mm)
+        max_transpiration : float
+            the maximum potential water lost to crop transpiration for the current day (mm)
+        water_distro_parameter : float
+            the water-use distribution parameter of the crop (unitless)
+        upper_depths : list[float]
+            depths to the top of each soil layer (mm)
+        lower_depths : list[float]
+            depths to the bottom of each soil layer (mm)
+
+        Returns
+        -------
+        potential_uptakes : list[float]
+            the crop's maximum potential water uptake for each soil layer (mm) during the current day
+
+        Raises
+        ------
+        Exception
+            If the lengths of `upper_depths` and `lower_depths` are not equal.
+        """
+        if len(upper_depths) != len(lower_depths):
+            raise Exception("upper_depths and lower_depths must be the same length")
+
+        potential_uptakes = []
+        for upper, lower in zip(upper_depths, lower_depths):
+            top_potential = WaterUptake._determine_max_water_uptake_to_depth(root_depth, upper, max_transpiration,
+                                                                             water_distro_parameter)
+            bottom_potential = WaterUptake._determine_max_water_uptake_to_depth(root_depth, lower, max_transpiration,
+                                                                                water_distro_parameter)
+            potential_uptakes.append(bottom_potential - top_potential)
+
+        return potential_uptakes
+
+    @staticmethod
+    def _determine_max_water_uptake_to_depth(root_depth: float, depth: float, max_transpiration: float,
+                                             water_distro_parameter: float) -> float:
+        """Calculate the amount of maximum amount water that can possibly be taken up by the plant under ideal
+        conditions.
+
+        SWAT Equation: 5:2.2.1
```

Comment:
```suggestion
        Reference
        -------
        SWAT Equation: 5:2.2.1
```
My PRs and some of Ed's PRs have been referencing like this, maybe it's better that we all stick to one way?